iroh/src/dns.rs

```diff
@@ -5,108 +5,22 @@
-/// interface is configured.  We do not want to use these DNS servers, the chances of them
-/// being usable are almost always close to zero, while the chance of DNS configuration
-/// **only** relying on these servers and not also being configured normally are also almost
-/// zero.  The chance of the DNS resolver accidentally trying one of these and taking a
-/// bunch of timeouts to figure out they're no good are on the other hand very high.
-const WINDOWS_BAD_SITE_LOCAL_DNS_SERVERS: [IpAddr; 3] = [
-    IpAddr::V6(Ipv6Addr::new(0xfec0, 0, 0, 0xffff, 0, 0, 0, 1)),
-    IpAddr::V6(Ipv6Addr::new(0xfec0, 0, 0, 0xffff, 0, 0, 0, 2)),
-    IpAddr::V6(Ipv6Addr::new(0xfec0, 0, 0, 0xffff, 0, 0, 0, 3)),
-];
-
-/// Get resolver to query MX records.
-///
-/// We first try to read the system's resolver from `/etc/resolv.conf`.
-/// This does not work at least on some Androids, therefore we fallback
-/// to the default `ResolverConfig` which uses eg. to google's `8.8.8.8` or `8.8.4.4`.
-fn create_default_resolver() -> Result<TokioResolver> {
-    let (system_config, mut options) =
-        hickory_resolver::system_conf::read_system_conf().unwrap_or_default();
-
-    // Copy all of the system config, but strip the bad windows nameservers.  Unfortunately
-    // there is no easy way to do this.
-    let mut config = hickory_resolver::config::ResolverConfig::new();
-    if let Some(name) = system_config.domain() {
-        config.set_domain(name.clone());
-    }
-    for name in system_config.search() {
-        config.add_search(name.clone());
-    }
-    for nameserver_cfg in system_config.name_servers() {
-        if !WINDOWS_BAD_SITE_LOCAL_DNS_SERVERS.contains(&nameserver_cfg.socket_addr.ip()) {
-            config.add_name_server(nameserver_cfg.clone());
-        }
-    }
-
-    // see [`ResolverExt::lookup_ipv4_ipv6`] for info on why we avoid `LookupIpStrategy::Ipv4AndIpv6`
-    options.ip_strategy = hickory_resolver::config::LookupIpStrategy::Ipv4thenIpv6;
-
-    let resolver = Resolver::tokio(config, options);
-    Ok(resolver)
+pub fn resolver() -> &'static DnsResolver {
+    default_resolver()
 }
 
 /// Extension trait to [`DnsResolver`].
 pub trait ResolverExt {
-    /// Perform an ipv4 lookup with a timeout.
-    fn lookup_ipv4<N: IntoName>(
-        &self,
-        host: N,
-        timeout: Duration,
-    ) -> impl Future<Output = Result<impl Iterator<Item = IpAddr>>>;
-
-    /// Perform an ipv6 lookup with a timeout.
-    fn lookup_ipv6<N: IntoName>(
-        &self,
-        host: N,
-        timeout: Duration,
-    ) -> impl Future<Output = Result<impl Iterator<Item = IpAddr>>>;
-
-    /// Race an ipv4 and ipv6 lookup with a timeout.
-    fn lookup_ipv4_ipv6<N: IntoName + Clone>(
-        &self,
-        host: N,
-        timeout: Duration,
-    ) -> impl Future<Output = Result<impl Iterator<Item = IpAddr>>>;
-
     /// Looks up node info by DNS name.
     fn lookup_by_name(&self, name: &str) -> impl Future<Output = Result<NodeAddr>>;
```

Comment:
why not move the others as well?

Reply:
Went forward with this and moved the node_info module and the lookup methods for nodes to iroh_relay as well.

This means iroh_relay now has a dependency on pkarr (because the functions to parse node info from pkarr packets live on the same structs as parsing from hickory dns records) - without default features though, so should be fine (includes neither the relay nor the dht clients).